frontend-react/src/components/Admin/AdminReceiverDashboard.tsx

```diff
@@ -290,79 +302,66 @@ class TimeSlots implements IterateTimeSlots {
  * build the dictionary with a special path+key
  * @param dataIn
  */
-function makeDictionary(dataIn: AdmConnStatusDataType[]): DataDictionary {
-    // empty case
-    if (Object.keys(dataIn).length === 0) {
-        return {};
-    }
-
-    const data: DataDictionary = {};
-    let needsSorting = false;
-
-    let lastkey = "";
-    for (let item of dataIn) {
-        const key = `${item.organizationName}|${item.receiverName}\t${item.connectionCheckStartedAt}`;
-        data[key] = item;
-        if (lastkey > key) {
-            needsSorting = true;
+const sortStatusData = (
+    dataIn: AdmConnStatusDataType[]
+): AdmConnStatusDataType[] => {
+    const strcmp = (a: string, b: string) => (a < b ? -1 : a > b ? 1 : 0);
+
+    const sortCompare = (
```

Comment:
Yeah, I tend to tightly scope EVERYTHING even when I don't need to do so or it makes it harder to read. I'll move it out.  It will make it easier to unit test, too.